proofs/eo/cpc/rules/Sets.eo

```diff
@@ -30,3 +30,51 @@
   :requires ((($set_is_empty_eval t) b))
   :conclusion (= (set.is_empty t) b)
 )
+
+; rule: sets_singleton_inj
+; implements: ProofRule::SETS_SINGLETON_INJ
+; premises:
+; - deq: An equality between singleton sets.
+; conclusion: The elements of the singleton sets are equal.
+(declare-rule sets_singleton_inj ((T Type) (a T) (b T))
+  :premises ((= (set.singleton a) (set.singleton b)))
+  :conclusion (= a b)
+)
+
+; rule: sets_ext
+; implements: ProofRule::SETS_EXT
+; premises:
+; - deq: A disequality between sets.
+; conclusion: The sets are disequal for a witness lement, as given by the skolem (@set_deq_diff a b).
+(declare-rule sets_ext ((T Type) (a (Set T)) (b (Set T)))
+  :premises ((not (= a b)))
+  :conclusion (not (= (set.member (@sets_deq_diff a b) a) (set.member (@sets_deq_diff a b) b)))
+)
+
+
+; rule: sets_filter_down
+; implements: ProofRewriteRule::SETS_FILTER_DOWN
+; premises:
+; - mem: A membership of element x into a set filter term.
+; conclusion: >
```

Comment:
```suggestion
; conclusion: 
```

Reply:
This is used to do multi-line comments based on the doc style.